src/controllers/dynakube/dynakube_controller.go

```diff
@@ -211,6 +214,28 @@ func (controller *Controller) setupIstio(ctx context.Context, dynakube *dynatrac
 	return istioReconciler, nil
 }
 
+func (controller *Controller) createDynatraceRegistryClient(ctx context.Context, dynakube *dynatracev1beta1.DynaKube) (registry.ImageGetter, error) {
+	keychain := dynakube.PullSecretWithoutData()
```

Comment:
```suggestion
	pullSecret := dynakube.PullSecretWithoutData()
```
`keychain` is a bit confusing name, as we have `Keychain` as a specific part of the registry.
Also when saying `registry.WithKeyChainSecret(&keychain)` reads a bit backwards, like "the Secret for the Keychain is the Keychain it self". `registry.WithKeyChainSecret(&pullSecret)` read much nicer IMO

Reply:
fixed in c247cec7